docs/fundamentals/gateway/custom-gateway.md

```diff
@@ -0,0 +1,326 @@
+* name: gateway pod name.
+* logger: Jina logger object.
+* tracing: whether runtime tracing is enabled or not.
+* tracer_provider: OpenTelemetry `TraceProvider` object.
+* streamer: {class}`~jina.serve.streamer.GatewayStreamer`. Use this object to send Documents from the Gateway to Executors. Refer to {ref}`this section <gateway-streamer>` for more information.
+* runtime_args: `argparse.Namespace` object containing runtime arguments.
+* port: main port exposed by the Gateway.
+* ports: list all ports exposed by the Gateway.
+* protocols: list all protocols supported by the Gateway.
+* host: host address to which the Gateway server should be bound.
+
+```{admonition} Nonte
+:class: note
+
+Jina provides the Gateway with a list of ports and protocols to expose. Therefore, a Custom Gateway can serve on 
+multiple ports and protocols.
+```
+
+### User-defined parameters
+Users can add other parameters by implementing a constructor `__init__`. Parameters of the constructor can be set and 
+overridden in the Flow Python API (using `uses_with` parameter) or in the YAML configuration when including the gateway.
+Refer to the {ref}`Use Custom Gateway section <use-custom-gateway>` for more information.
+
+(gateway-streamer)=
+## Calling Executors with {class}`~jina.serve.streamer.GatewayStreamer`
+{class}`~jina.serve.streamer.GatewayStreamer` allows you to interface with Executors within the gateway. An instance of 
+this class knows about the Flow structure and where each Executor lives. You can get this 
+object in 2 different ways:
+* A `streamer` object (instance of {class}`~jina.serve.streamer.GatewayStreamer`) is injected by Jina to your gateway class.
+* In case your server logic cannot access the Gateway class (for instance separate script), you can still get a streamer 
+object using {meth}`~jina.serve.streamer.get_streamer()`.
+
+After transforming requests that arrive to the gateway server into Documents, you can send them to Executors in the Flow 
+using {meth}`~jina.serve.streamer.GatewayStreamer.stream_docs` :
+```python
+from jina.serve.runtimes.gateway.http.fastapi import FastAPIBaseGateway
+from jina import Document, DocumentArray
+from fastapi import FastAPI
+
+
+class MyGateway(FastAPIBaseGateway):
+    @property
+    def app(self):
+        app = FastAPI()
+
+        @app.get("/")
+        def get(text: str):
+            result = None
+            async for docs in self.streamer.stream_docs(
+                docs=DocumentArray([Document(text=text)]),
+                exec_endpoint='/',
+            ):
+                result = docs[0].text
+            return {'result': result}
+```
+
+
+## Required health-checks
+Jina relies on performing health-checks to determine the health of the gateway. In environments like kubernetes, 
+docker-compose and Jina Cloud, this information is crucial to orchestrate the Gateway.
+Since the user has the full power over custom gateways, he always has the responsibility of implementing health-check 
```

Comment:
Gender!!1!
"Since you have full control over your custom gateways, you are always responsible for implementing health-check endpoints."